kafka-ui-api/src/main/java/com/provectus/kafka/ui/kafka/KafkaService.java

```diff
@@ -0,0 +1,107 @@
+package com.provectus.kafka.ui.kafka;
+
+import com.provectus.kafka.ui.cluster.model.MetricsConstants;
+import com.provectus.kafka.ui.cluster.model.KafkaCluster;
+import com.provectus.kafka.ui.model.Partition;
+import com.provectus.kafka.ui.model.Replica;
+import com.provectus.kafka.ui.model.Topic;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.I0Itec.zkclient.ZkClient;
+import org.apache.kafka.clients.admin.*;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.TopicPartitionInfo;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+import static com.provectus.kafka.ui.cluster.model.MetricsConstants.CLUSTER_ID;
+import static com.provectus.kafka.ui.cluster.model.MetricsConstants.ZOOKEEPER_STATUS;
+
+@Service
+@RequiredArgsConstructor
+public class KafkaService {
+
+    @SneakyThrows
+    public void loadClusterMetrics(KafkaCluster kafkaCluster) {
+        isZookeeperRunning(kafkaCluster);
+
+        Properties properties = new Properties();
+        properties.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaCluster.getBootstrapServers());
+        properties.put(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG, 5000);
+        AdminClient adminClient = AdminClient.create(properties);
+
+        loadMetrics(kafkaCluster, adminClient);
+        loadTopics(kafkaCluster, adminClient);
+    }
+
+    @SneakyThrows
+    private void loadTopics(KafkaCluster kafkaCluster, AdminClient adminClient) {
+        ListTopicsOptions listTopicsOptions = new ListTopicsOptions();
+        listTopicsOptions.listInternal(true);
+        Collection<String> topicListings = adminClient.listTopics(listTopicsOptions).names().get();
+
+        DescribeTopicsResult topicDescriptionsWrapper= adminClient.describeTopics(topicListings);
+        Map<String, KafkaFuture<TopicDescription>> topicDescriptionFuturesMap = topicDescriptionsWrapper.values();
+        List<Topic> foundTopics = new ArrayList<>();
+        String clusterId = kafkaCluster.getMetricsMap().get(CLUSTER_ID);
+        for (KafkaFuture<TopicDescription> value : topicDescriptionFuturesMap.values()) {
+            TopicDescription topicDescription = value.get();
```

Comment:
If any topic will fail, the whole operation will fail, segregate it in separate operations.

Reply:
fixed, if can't get data from kafka, logging exception and going to next topic